jetty-server/src/main/java/org/eclipse/jetty/server/AsyncContextEvent.java

```diff
@@ -25,25 +25,28 @@
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 
+import org.eclipse.jetty.http.HttpURI;
 import org.eclipse.jetty.server.handler.ContextHandler.Context;
 import org.eclipse.jetty.util.thread.Scheduler;
 
 public class AsyncContextEvent extends AsyncEvent implements Runnable
 {
     private final Context _context;
     private final AsyncContextState _asyncContext;
+    private final HttpURI _providedUri;
     private volatile HttpChannelState _state;
     private ServletContext _dispatchContext;
     private String _dispatchPath;
     private volatile Scheduler.Task _timeoutTask;
     private Throwable _throwable;
 
-    public AsyncContextEvent(Context context, AsyncContextState asyncContext, HttpChannelState state, Request baseRequest, ServletRequest request, ServletResponse response)
+    public AsyncContextEvent(Context context, AsyncContextState asyncContext, HttpChannelState state, Request baseRequest, ServletRequest request, ServletResponse response, boolean provided)
     {
         super(null, request, response, null);
         _context = context;
         _asyncContext = asyncContext;
         _state = state;
+        _providedUri = provided ? new HttpURI(baseRequest.getHttpURI()) : null;
```

Comment:
This new `boolean` parameter requires more brainpower to understand than I'm comfortable with. IMHO it should either be renamed (`useProvidedUri`?) or documented, or maybe both.

Reply:
I'll pass in the URI and call it baseURI.